src/promptflow/tests/sdk_cli_test/conftest.py

```diff
@@ -182,25 +190,29 @@ def mock_invoke_tool(self, func, *args, **kwargs):
 
 
 @pytest.fixture
-def recording_enabled(mocker: MockerFixture):
+def recording_file_override(request: pytest.FixtureRequest, mocker: MockerFixture):
+    if RecordStorage.is_replaying_mode() or RecordStorage.is_recording_mode():
+        lowercase_cls_name = request.cls.__name__.lower()
+        file_path = RECORDINGS_TEST_CONFIGS_ROOT / f"{lowercase_cls_name}_node_cache.shelve"
```

Comment:
Can we use single binary file? Is separating to multilple files due to our tests could be running in parallel?
Add some comments on this

Reply:
let me investigate